emulators/noah/Portfile

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup               GitHub 1.0
+
+name    noah
+subport noahstrap {}
+
+categories              emulators
+maintainers             nomaintainer
+platforms               darwin
+
+license                 Dual MIT/GPL
+description             Linux ABI implementation for OSX
+
+long_description        Noah is a Darwin subsystem for Linux, or\
+                        \"Bash on Ubuntu on Mac OS X\". Noah is implemented as a hypervisor \
+                        that traps linux system calls and translates them into Darwin's \
+                        system calls. Noah also has an interpreter of ELF files so that \
+                        binary executables of Linux run directly and flawlessly without \
+                        any modifications. i.e. it's effectively an OSX Linux Execution \
+                        Flavour, similar to that of FreeBSD Linuxolator, aka Linux \
+                        Emulation, aka Linux ABI. In other words, it's the exact reverse \
+                        of the Linux Darling project: https://github.com/darlinghq.
+
+# others do not make sense at this point
+supported_archs         x86_64
+
+if {${subport} eq "noah"} {
+    PortGroup               cmake 1.1
+    depends_build-append    port:noahstrap
+    github.setup            linux-noah noah 0.5.1
+    checksums               rmd160  acfb387976bd21a0b2a02a1490a4ef1891d73103 \
+                            sha256  6a2ecba330aa6b3aacc7a2799f4d4a2fbdb5413dd3a8aa9553d3bcd1a81424c0 \
+                            size    49645428
+
+} elseif {${subport} eq "noahstrap"} {
+
+    depends_build-append    port:pv \
+                            port:gnutar \
+                            port:perl5.26
+                            
+    github.setup            linux-noah noahstrap 2.0.0
+    checksums               rmd160 7eb7168d4211255e3ff6e603b68d36a1746ecda8 \
+                            sha256  616a05c9eb3c12bfcfe12933652a27663c56c46fe91b5d799061742e0973f3fe \
+                            size    1946
+
+    post-extract  {
+        reinplace "s|/usr/bin/perl|${prefix}/bin/perl5.26|g" ${worksrcpath}/noahstrap
```

Comment:
Now MacPorts wants us to hardcode in a specific PERL. I just have to think a bit more about this. `noahstrap` will be run later on, won't it? So this dependency on perl5.26 will need to be fulfilled exactly when ? Not necessarily at build time for noahstrap, as I have here. It's really at run time of `noahstrap`. Also - at run time of `noah`? This one is a bit complicated to think through. I think perl5.26 might need to be a run dep for both `noah` and `noahstrap` ...

Reply:
OK. So we could make `perl5.26` a runtime dep of `noah` then, and it should all work out. I'm also pondering whether `noahstrap` really needs to be a separate port, or whether we might roll it into `noah` itself as part of `noah`, if it's not used at any other time. There is a way to do that as well, if we decide to go with that idea...

Reply:
I also thought of that, but I am no macports guru, so combining two projects from GitHub when building is a bit of an overkill to me. (The way how I imagine it brings far too many variables, like separate build directories and staging directories and so on).